models/base/scratch/snowflake/snowplow_media_player_base_events_this_run.sql

```diff
@@ -0,0 +1,115 @@
+{{
+  config(
+    tags=["this_run"],
+    sql_header=snowplow_utils.set_query_tag(var('snowplow__query_tag', 'snowplow_dbt'))
+  )
+}}
+
+{%- set lower_limit, upper_limit = snowplow_utils.return_limits_from_model(ref('snowplow_media_player_base_sessions_this_run'),
+                                                                          'start_tstamp',
+                                                                          'end_tstamp') %}
+
+-- check for exceptions
+{% if var("snowplow__enable_whatwg_media") is false and var("snowplow__enable_whatwg_video") %}
+  {{ exceptions.raise_compiler_error("variable: snowplow__enable_whatwg_video is enabled but variable: snowplow__enable_whatwg_media is not, both need to be enabled for modelling html5 video tracking data.") }}
+{% elif not var("snowplow__enable_youtube") and not var("snowplow__enable_whatwg_media") %}
+  {{ exceptions.raise_compiler_error("No media context enabled. Please enable as many of the following variables as required: snowplow__enable_youtube, snowplow__enable_whatwg_media, snowplow__enable_whatwg_video") }}
+{% endif %}
+
+with prep as (
+
+  select
+
+    a.event_id,
+    a.contexts_com_snowplowanalytics_snowplow_web_page_1[0]:id::varchar as page_view_id,
+    a.domain_sessionid,
+    b.domain_userid,
+    a.page_referrer,
+    a.page_url,
+    a.geo_region_name,
+    a.br_name,
+    a.dvce_type,
+    a.os_name,
+    a.os_timezone,
+    a.derived_tstamp as start_tstamp,
+    a.collector_tstamp,
+
+    -- unpacking the media player event
+    a.unstruct_event_com_snowplowanalytics_snowplow_media_player_event_1:label::varchar as media_label,
+    a.unstruct_event_com_snowplowanalytics_snowplow_media_player_event_1:type::varchar as event_type,
+
+    -- unpacking the media player object
+    round(a.contexts_com_snowplowanalytics_snowplow_media_player_1[0]:duration::int) as duration,
+    a.contexts_com_snowplowanalytics_snowplow_media_player_1[0]:currentTime::float as player_current_time,
```

Comment:
Being picky, but if this is being cast as an int, what is the point of rounding it?

Reply:
Have changed the casting to float before rounding here and also for BigQuery. For other warehouses, it was already cast to float before rounding.